browser/views.py

```diff
@@ -686,3 +685,51 @@ def delete_investigator_record(request, person_id, research_id):
         Investigator.objects.filter(id=research_id).delete()
 
     return HttpResponseRedirect(reverse('person', args=(person_id,)))
+
+
+@staff_member_required
+def position(request, person_id, position_id=None):
+    person = get_object_or_404(Person, pk=person_id)
+    template = "browser/position.html"
+
+    context = {
+        'person': person,
+        'position_id': position_id
+    }
+
+    if request.method == 'GET':
+        if position_id:
+            position = get_object_or_404(Position, pk=position_id)
+            form = PositionForm(initial={'subject': position.subject, 'role': position.role,
+                                         'year': position.year, 'start_date': position.start_date,
+                                         'end_date': position.end_date})
+        else:
+            form = PositionForm()
+
+        context.update({
+            'form': form
+        })
+
+    if request.method == 'POST':
+        form = PositionForm(request.POST)
+        if form.is_valid():
+            if position_id is None:
+                Position.objects.create(
+                    person_id=person_id,
+                    changed_by=request.user,
+                    **form.cleaned_data
+                )
+            else:
+                Position.objects.filter(id=position_id).update(**form.cleaned_data)
+            return HttpResponseRedirect(reverse('person', args=(person.id,),)+'?person-positions')
```

Comment:
this is not proper url format. Parameters are not passed like that.

Reply:
Changed passing of url parameters to the following:
```
('person', args=(person.id,),)+'?tab=person-positions')
```